### PR TITLE
Kantis/service refactor

### PR DESCRIPTION
**Pull request by Kantis**
Builds on #305 and #307

Extracts all state into a Service as recommended in the [guidelines](https://plugins.jetbrains.com/docs/intellij/plugin-components.html#manage-state). 

Running scans and stuff on background thread.

### CODE REVIEW

**Comment by Kantis** [comment]
Seems like Services using coroutines aren't supported for IC-223... Not sure if we can drop that version as well?

File: src/main/kotlin/io/kotest/plugin/intellij/toolwindow/KotestTestExplorerService.kt [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+   interface ModelListener {
+      fun setModel(treeModel: TreeModel)
+   }
+
+   private val modelListeners = mutableListOf<ModelListener>()
+   fun registerModelListener(modelListener: ModelListener) { modelListeners.add(modelListener) }
+
+   private fun reloadModelInBackgroundThread() {
+      scope.launch {
+         withContext(Dispatchers.Default) {
```
Comment (sksamuel) [inline src/main/kotlin/io/kotest/plugin/intellij/toolwindow/KotestTestExplorerService.kt:66]: can you just pass the dispatcher to the launch method ?

**Review by sksamuel — APPROVED** [review]

**Comment by sksamuel** [comment]
> Seems like Services using coroutines aren't supported for IC-223... Not sure if we can drop that version as well?
> 
> edit: I figured a way around it for 223.. but would be nice to get rid of workarounds later.

223 is still pretty popular, should keep it for remainder of 2024 I'd say

**Comment by Kantis** [comment]
> > Seems like Services using coroutines aren't supported for IC-223... Not sure if we can drop that version as well?
> > edit: I figured a way around it for 223.. but would be nice to get rid of workarounds later.
> 
> 223 is still pretty popular, should keep it for remainder of 2024 I'd say

Alright 👍🏼  are there stats for the versions somewhere?

**Pull request merged**